### PR TITLE
Allowing to enabled CPU_PIN on other oses than Linux (FreeBSD at the …

### PR DESCRIPTION
**Pull request by devnexen**
…moment tough).

New internal api `_iso_getcpu` abstraction, for now working on amd64 arch
for non Linux.

### CODE REVIEW

File: src/iso_alloc.c [hunk 1]
```diff
@@ -2117,3 +2117,23 @@ EXTERNAL_API iso_alloc_root *_get_root(void) {
...
+ * the cost of feature's abstraction here.
+ */
+INTERNAL_HIDDEN INLINE int _iso_getcpu(void) {
+#if defined(SCHED_GETCPU)
+    return sched_getcpu();
+#elif defined(__x86_64__)
+    /* rdtscp is not always available and is pretty slow */
+    unsigned int a;
+    const unsigned int cpunodesegment = 15 * 8 + 3;
+    __asm__ volatile("lsl %1, %0" : "=r"(a) : "r"(cpunodesegment));
```
Comment (struct) [inline src/iso_alloc.c:2133]: Can you add a comment explaining how the LSL instructions works here?

**Comment by struct** [comment]
Were you able to successfully test this with `CPU_PIN` enabled?

**Comment by devnexen** [comment]
Yes I did tested, with the rdtscp instruction, tests took forever.

**Comment by struct** [comment]
Yes the `CPU_PIN` functionality in general incurs a non-trivial perf hit.

**Pull request merged**